packages/@aws-cdk/custom-resource-handlers/lib/custom-resources/aws-custom-resource-handler/index.ts

```diff
@@ -1,17 +1,277 @@
+    },
+  };
+
+  return new Promise((resolve, reject) => {
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-require-imports
+      const request = require('https').request(requestOptions, resolve);
+      request.on('error', reject);
+      request.write(responseBody);
+      request.end();
+    } catch (e) {
+      reject(e);
+    }
+  });
 }
 
-export function v3handler(event: AWSLambda.CloudFormationCustomResourceEvent, context: AWSLambda.Context) {
-  return require('./aws-sdk-v3-handler').handler(event, context);
+function decodeCall(call: string | undefined) {
+  if (!call) { return undefined; }
+  return JSON.parse(call);
 }
 
-export function handler(event: AWSLambda.CloudFormationCustomResourceEvent, context: AWSLambda.Context) {
-  const env = process.env.AWS_EXECUTION_ENV;
-  return (env && env >= 'AWS_Lambda_nodejs18.x'
-    ? v3handler(event, context)
-    : v2handler(event, context));
+function startsWithOneOf(searchStrings: string[]): (string: string) => boolean {
+  return function(string: string): boolean {
+    for (const searchString of searchStrings) {
+      if (string.startsWith(searchString)) {
+        return true;
+      }
+    }
+    return false;
+  };
+}
+
+/* eslint-disable @typescript-eslint/no-require-imports, import/no-extraneous-dependencies */
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent, context: AWSLambda.Context) {
+  try {
+    event.ResourceProperties.Create = decodeCall(event.ResourceProperties.Create);
+    event.ResourceProperties.Update = decodeCall(event.ResourceProperties.Update);
+    event.ResourceProperties.Delete = decodeCall(event.ResourceProperties.Delete);
+    let data: { [key: string]: string } = {};
+
+    // Default physical resource id
+    let physicalResourceId: string;
+    switch (event.RequestType) {
+      case 'Create':
+        physicalResourceId = event.ResourceProperties.Create?.physicalResourceId?.id ??
+                             event.ResourceProperties.Update?.physicalResourceId?.id ??
+                             event.ResourceProperties.Delete?.physicalResourceId?.id ??
+                             event.LogicalResourceId;
+        break;
+      case 'Update':
+      case 'Delete':
+        physicalResourceId = event.ResourceProperties[event.RequestType]?.physicalResourceId?.id ?? event.PhysicalResourceId;
+        break;
+    }
+    const call: AwsSdkCall | undefined = event.ResourceProperties[event.RequestType];
+    // if there is a call there will always be logging configured -- otherwise, in the event of no call, logging
+    // wasn't configured so just default to existing behavior
+    const logApiResponseData = call?.logApiResponseData ?? true;
+    if (call) {
+      const apiCall = new ApiCall(call.service, call.action);
+
+      let awsSdk: AwsSdk | Promise<AwsSdk> = loadAwsSdk(apiCall.v3PackageName, event.ResourceProperties.InstallLatestAwsSdk);
```

Comment:
Yeah, I agree that this is weird. I've updated `loadAwsSdk` to explicitly return `Promise<AwsSdk>`. I also updated this line to not have an explicit type since this will be picked up by the function return type.